### PR TITLE
document why Redeemers and TxDats use MemoBytes

### LINKED ISSUE #3236: `Redeemers` and `MemoBytes`
Investigate why `Redeemers` are based on `MemoBytes`, it doesn't seem like they need it.

If this is right then we should remove `MemoBytes` for `Redeemers`, since that only introduces unnecessary overhead.

### PR DESCRIPTION
**Pull request by JaredCorduan**
resolves #3236

# Description

Add your description here, if it fixes a particular issue please provide a
[link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=)
to the issue.

# Checklist

- [x] Commit sequence broadly makes sense
- [x] Commits have useful messages
- [x] New tests are added if needed and existing tests are updated
- [x] Any changes are noted in the [changelog](https://github.com/input-output-hk/cardano-ledger/blob/master/CHANGELOG.md)
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (which can be run with `scripts/fourmolize.sh`
- [x] Self-reviewed the diff

### CODE REVIEW

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWits.hs [hunk 1]
```diff
@@ -144,6 +144,10 @@ instance Typeable era => ToCBOR (RedeemersRaw era) where
 instance Memoized Redeemers where
   type RawType Redeemers = RedeemersRaw
 
+-- | Note that 'Redeemers' are based on 'MemoBytes' since we must preserve
+-- the original bytes for the 'ScriptIntegrity'.
```
Comment (lehins) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWits.hs]: In order for haddock to properly link it
```suggestion
-- the original bytes for the 'Cardano.Ledger.Alonzo.Tx.ScriptIntegrity'.
```

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWits.hs [hunk 2]
```diff
@@ -278,6 +282,10 @@ nullDats (TxDats' d) = Map.null d
 instance (Era era) => FromCBOR (Annotator (TxDatsRaw era)) where
   fromCBOR = decode $ fmap (TxDatsRaw . keyBy hashData) <$> listDecodeA From
 
+-- | Note that 'TxDats' are based on 'MemoBytes' since we must preserve
+-- the original bytes for the 'ScriptIntegrity'.
```
Comment (lehins) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWits.hs]: Same here
```suggestion
-- the original bytes for the 'Cardano.Ledger.Alonzo.Tx.ScriptIntegrity'.
```

**Review by lehins — APPROVED** [review]
:+1:

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWits.hs (hunk 1 above)
Comment (JaredCorduan) [reply eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWits.hs]: thanks!

**Pull request merged**